redisearch/client.py

```diff
@@ -38,7 +40,12 @@ def append_arg(self, value):
         self.args.append(value)
 
     def redis_args(self):
```

Comment:
You could setattr the args on the class if not set. Since it seems like you use it once.  Yes, you'll have to change code, but things will run a lot faster. Something like:

if getattr(self, "redis_args", None):
 ### do your work to set everything
 self.redis_args = <the new list you created>

If this is called repeatedly you get a speedup. If users one off the object (i.e no reuse) then ignore this setattr piece and instead turn this into a property for people to use, having the function return the property so as not to break the API. Then, we can mark redis_args as [depreciated](https://stackoverflow.com/questions/2536307/decorators-in-the-python-standard-lib-deprecated-specifically).

Reply:
But what if someone used the `append_arg` and changed the `args`?

I think I can use `delattr` in this case, but I'm not sure how we can mark `redis_args` as deprecated.